src/blockchain/web3.ts

```diff
@@ -35,3 +41,75 @@ export const web3Ready$ = web3Status$.pipe(filter(status => status === 'ready'))
 export function setupFakeWeb3ForTesting() {
   web3 = new Web3();
 }
+
+export const getCurrentProviderName = (provider = (window as Web3Window).web3.currentProvider) => {
```

Comment:
what's the return type of this fn? Can we be explicit and create interface for it?